test/stateless/stateless-pipeline.test.ts

```diff
@@ -1,37 +1,42 @@
-import { App, Aspects } from 'aws-cdk-lib';
+import { App, Aspects, Stack } from 'aws-cdk-lib';
 import { Annotations, Match } from 'aws-cdk-lib/assertions';
 import { SynthesisMessage } from 'aws-cdk-lib/cx-api';
 import { AwsSolutionsChecks, NagSuppressions } from 'cdk-nag';
 import { StatelessPipelineStack } from '../../lib/pipeline/orcabus-stateless-pipeline-stack';
 
+// we are mocking the deployment stack here, as we have a dedicated cdk-nag test for deployment stack
+// see the ./stateless-deployment.test.ts
+jest.mock('../../lib/workload/orcabus-stateless-stack', () => {
+  return {
+    OrcaBusStatelessStack: jest.fn().mockImplementation((value) => {
+      return new Stack(value, 'mockStack', {});
+    }),
+  };
+});
+
 function synthesisMessageToString(sm: SynthesisMessage): string {
   return `${sm.entry.data} [${sm.id}]`;
 }
 
 // Stateless Stack
 describe('cdk-nag-stateless-pipeline', () => {
-  let stack: StatelessPipelineStack;
-  let app: App;
-
-  beforeEach(() => {
-    app = new App({});
-    stack = new StatelessPipelineStack(app, 'TestStack', {
-      env: {
-        account: '123456789',
-        region: 'ap-southeast-2',
-      },
-    });
-    Aspects.of(stack).add(new AwsSolutionsChecks());
-
-    NagSuppressions.addStackSuppressions(stack, [
-      { id: 'AwsSolutions-IAM4', reason: 'Allow CDK Pipeline' },
-      { id: 'AwsSolutions-IAM5', reason: 'Allow CDK Pipeline' },
-      { id: 'AwsSolutions-S1', reason: 'Allow CDK Pipeline' },
-      { id: 'AwsSolutions-KMS5', reason: 'Allow CDK Pipeline' },
-      { id: 'AwsSolutions-CB3', reason: 'Allow CDK Pipeline' },
-    ]);
+  const app: App = new App({});
+  const stack: StatelessPipelineStack = new StatelessPipelineStack(app, 'TestStack', {
+    env: {
+      account: '123456789',
+      region: 'ap-southeast-2',
+    },
   });
 
+  Aspects.of(stack).add(new AwsSolutionsChecks());
+  NagSuppressions.addStackSuppressions(stack, [
+    { id: 'AwsSolutions-IAM4', reason: 'Allow CDK Pipeline' },
+    { id: 'AwsSolutions-IAM5', reason: 'Allow CDK Pipeline' },
+    { id: 'AwsSolutions-S1', reason: 'Allow CDK Pipeline' },
+    { id: 'AwsSolutions-KMS5', reason: 'Allow CDK Pipeline' },
+    { id: 'AwsSolutions-CB3', reason: 'Allow CDK Pipeline' },
+  ]);
+
   test('cdk-nag AwsSolutions Pack errors', () => {
     const errors = Annotations.fromStack(stack)
```

Comment:
@mmalenic I didn't look into detailed to the implementation of the `fromStack` that forces to multiple build on every call. But just from the build log, the build only happen once despite the `fromStack` function being called multiple times.

Reply:
Nice! I guess there's some build caching going on?

The Rust code in filemanager is compiling multiple times anyway even with a normal `cdk synth` (i.e. outside the context of parallel test execution) - I'll look into why this is happening, and I don't think it's related to the tests.